django/contrib/admin/options.py

```diff
@@ -613,8 +614,14 @@ def get_form(self, request, obj=None, **kwargs):
         # if exclude is an empty list we pass None to be consistent with the
         # default on modelform_factory
         exclude = exclude or None
+
+        # Cancel out overridden form fields which are in readonly_fields.
+        new_attrs = OrderedDict([(f, None) for f in readonly_fields
```

Comment:
> I don't think keeping the attributes as None on the modelform is the right approach. 

It's the official way of un-defining overridden model form fields, if i understand ticket [#8620](https://code.djangoproject.com/ticket/8620) and commit 9fb0f5dddc4cf7f2d294af1bcde2c359cffd90a5 correctly.

> I think they shouldn't be there at all.

How would you suggest we do this ?